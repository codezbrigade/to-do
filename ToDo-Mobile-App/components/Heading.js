import React, { useEffect } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { COLORS, FONTS } from '../constants';

import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';

import { RFValue } from 'react-native-responsive-fontsize';

const Heading = () => (
  <View style={styles.headingContainer}>
    <Text style={styles.heading}>Any TO_DO</Text>
  </View>
);


export default Heading;

const styles = StyleSheet.create({
  headingContainer: {
    marginVertical: hp(1.8),
  },
  heading: {
    fontSize: RFValue(32),
    fontFamily: FONTS.LeelawadeeUI,
    lineHeight: hp(6),
    fontWeight: '700',
    color: COLORS.main
  }
})