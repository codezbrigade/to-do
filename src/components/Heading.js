import React from 'react';
import {Text, View} from 'react-native';

import {styles} from './Heading.styles';

const Heading = ({styleProps}) => (
  <View style={[styles.headingContainer, styleProps]}>
    <Text style={styles.heading}>Any TO_DO</Text>
  </View>
);

export default Heading;
