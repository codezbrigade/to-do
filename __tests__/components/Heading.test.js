import React from 'react';
import renderer from 'react-test-renderer';
import Heading from '../../AnyTODO/components/Heading';


describe('Heading component basics', () => {
  const tree = renderer.create(<Heading />).toJSON();

  beforeEach(() => {
    /* Runs before each test */
    console.log('_____________________');
  });

  test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Should have one parrent node and that should be View', () => {
    expect(tree.type).toEqual('View');
    expect(tree.type).toEqual('View');
  });

  test('Should have Heading text', () => {
    expect(tree.type).toEqual('View');
  });

  console.log(JSON.stringify(tree));
});