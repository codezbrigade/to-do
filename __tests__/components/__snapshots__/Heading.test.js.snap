// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component basics renders correctly 1`] = `
<View
  style={
    [
      {
        "marginVertical": 24,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#2E3192",
        "fontFamily": "LeelawadeeUI",
        "fontSize": 63,
        "fontWeight": "700",
        "lineHeight": 80,
      }
    }
  >
    Any TO_DO
  </Text>
</View>
`;
